king_phisher/client/windows/plugin_manager.py

```diff
@@ -229,42 +340,156 @@ def signal_renderer_toggled(self, _, path):
+						return
+			self.catalog_plugins.install_plugin(catalog_model[0], repo_model[0], plugin_model[0], DEFAULT_PLUGIN_PATH)
+			self.config['plugins.installed'][plugin_model[0]] = [catalog_model[0], repo_model[0]]
+			plugin_model[2] = True
+			plugin_model[6] = self.catalog_plugins.get_collection(catalog_model[0], repo_model[0])[plugin_model[0]]['version']
+			self.logger.info("installed plugin {} from catalog:{}, repository:{}".format(plugin_model[0], catalog_model[0], repo_model[0]))
+		self.application.plugin_manager.load_all(on_error=self._on_plugin_load_error)
+
+	def _disable_plugin(self, plugin_model):
+		self.application.plugin_manager.disable(plugin_model[0])
+		self.config['plugins.enabled'].remove(plugin_model[0])
+		plugin_model[1] = False
+
+	def _remove_matching_plugin(self, plugin_name, installed_plugin_info):
+		for catalog_model in self._model:
+			if catalog_model[0] != installed_plugin_info[0]:
+				continue
+			for repo_model in catalog_model.iterchildren():
+				if repo_model[0] != installed_plugin_info[1]:
+					continue
+				for plugin_model in repo_model.iterchildren():
+					if plugin_model[0] != plugin_name:
+						continue
+					if plugin_model[1]:
+						self._disable_plugin(plugin_model)
+					self._uninstall_plugin(self.catalog_plugins.get_collection(installed_plugin_info[0], installed_plugin_info[1]), plugin_model)
+					return True
 
-	def _set_plugin_info(self, name):
+	def _get_plugin_model_parents(self, plugin_model):
+		return plugin_model.parent, plugin_model.parent.parent
+
+	def _uninstall_plugin(self, plugin_collection, plugin_model):
+		for files in plugin_collection[plugin_model[0]]['files']:
+			file_name = files[0]
+			if os.path.isfile(os.path.join(DEFAULT_PLUGIN_PATH, file_name)):
+				os.remove(os.path.join(DEFAULT_PLUGIN_PATH, file_name))
+		del self.config['plugins.installed'][plugin_model[0]]
+		plugin_model[2] = False
+
+	def _set_plugin_info(self, model_instance):
 		stack = self.gobjects['stack_plugin_info']
 		textview = self.gobjects['textview_plugin_info']
 		buf = textview.get_buffer()
 		buf.delete(buf.get_start_iter(), buf.get_end_iter())
+		name = model_instance[0]
+		if model_instance[3] != 'Plugin':
+			stack.set_visible_child(textview)
+			self._set_non_plugin_info(model_instance)
+			return
 		if name in self._module_errors:
 			stack.set_visible_child(textview)
 			self._set_plugin_info_error(name)
 		else:
 			stack.set_visible_child(self.gobjects['grid_plugin_info'])
-			self._set_plugin_info_details(name)
+			self._set_plugin_info_details(model_instance)
+
+	def _set_non_plugin_info(self, model_instance):
+		textview = self.gobjects['textview_plugin_info']
+		buf = textview.get_buffer()
+		text = ''
+		if model_instance[3] == 'Catalog':
+			instance_information = self.catalog_plugins.catalogs[model_instance[0]]
+		else:
+			instance_information = self.catalog_plugins.get_repo(model_instance.parent[0], model_instance[0])
+
+		if 'title' in dir(instance_information):
```

Comment:
Just a couple lines before this you check `if model_instance[3] == 'Catalog'`. Can't you just do `text += "Catalog: ...` in that conditional because you'll know if it's a catalog or a repository.